modules/standards/test/standard-definition-authentication.test.ts

```diff
@@ -0,0 +1,186 @@
+  email: string;
+  password: string;
+};
+
+type PatientRequest = {
+  id: string;
+  name: string;
+  email: string;
+  password: string;
+};
+
+type PatientResponse = {
+  id: string;
+  name: string;
+  email: string;
+};
+
+type Credentials = {
+  email: string;
+  password: string;
+};
+
+type MyAuthSetting = {
+  credentials: Credentials;
+  options: {};
+};
+
+type MyEntityMap = {
+  patient: PlainPatient;
+};
+type MyGeneralReqResEntityMap = {
+  patient: {
+    request: PatientRequest;
+    response: PatientResponse;
+  };
+};
+
+type MemberSessionValue = { externalId: string; ttl: number };
+
+interface MyTypeMap extends GeneralTypeMap {
+  entities: MyGeneralReqResEntityMap;
+  customQueries: {};
+  customCommands: {};
+  auths: {
+    patient: MyAuthSetting;
+  };
+}
+
+const memoryClient = createEntityClient<MyEntityMap>();
+// TODO: need refinement
+const sessionClient = memoryClient.createSessionClient() as KvsClient<
+  Session<"patient", MemberSessionValue>
+>;
+
+class PatientDefinition extends StandardUserDefinition<
+  MyGeneralReqResEntityMap,
+  MyAuthSetting
+> {
+  constructor() {
+    super({
+      accountPropName: "email",
+      passwordPropName: "password",
+      entityClient: memoryClient,
+      ttl: 24 * 3600
+    });
+  }
+
+  authorize() {
+    return Promise.resolve(true);
+  }
+}
+
+const functionalGroup = {
+  customQueries: {},
+  customCommands: {},
+  users: {
+    patient: new PatientDefinition()
+  },
+  nonUsers: {}
+};
+
+let server: PhenylHttpServer<MyTypeMap>;
+before(() => {
+  const restApiHandler: PhenylRestApi<MyTypeMap> = new PhenylRestApi(
+    functionalGroup,
+    {
+      client: memoryClient,
+      sessionClient
+    }
+  );
+
+  server = new PhenylHttpServer(http.createServer(), {
+    restApiHandler
+  });
+  server.listen(8080);
+});
+
+after(() => {
+  server.close();
+});
+
+describe("Phenyl Standards with Authentication", async () => {
+  const client: PhenylHttpClient<MyTypeMap> = new PhenylHttpClient({
+    url: "http://localhost:8080"
+  });
+
+  // TODO: need refinement. Should preSession be created by others?
+  const preSession = await sessionClient.create({
+    entityName: "patient",
+    expiredAt: "",
+    userId: "shinout@example.com",
+    externalId: "",
+    ttl: 12345
+  });
+
+  let insertedEntity: PatientResponse;
+  it(" should be inserted first user", async () => {
+    const inserted = await client.insertAndGet(
+      {
+        entityName: "patient",
+        value: {
+          name: "Shin Suzuki",
+          email: "shinout@example.com",
+          password: "shin123"
+        }
+      },
+      preSession.id
+    );
+
+    insertedEntity = inserted.entity;
+    assert.strictEqual(insertedEntity.name, "Shin Suzuki");
+    assert.strictEqual(insertedEntity.email, "shinout@example.com");
+  });
+
+  let loggedInUser: PatientResponse;
+  let loggedInSession: Session;
+  it("can login by inserted user", async () => {
+    const logined = await client.login({
```

Comment:
[imo] logined is not a word. I think `loggedIn` . `patient`, or `loggedInPatient` would look nicer.